pkg/api/v1beta1/conversion.go

```diff
@@ -1112,4 +1112,26 @@ func init() {
 		// If one of the conversion functions is malformed, detect it immediately.
 		panic(err)
 	}
+
+	// Add field conversion funcs.
+	err = newer.Scheme.AddFieldLabelConversionFunc("v1beta1",
+		func(label, value string) (string, string, error) {
```

Comment:
I think this approach could cause labels from different objects to collide. Note that it's not just pods that need this. See pkg/registry/event/rest.go for another example.